zerver/tests/test_scheduled_messages.py

```diff
@@ -728,3 +728,77 @@ def test_attachment_handling(self) -> None:
             [scheduled_message.id],
         )
         self.assertEqual(scheduled_message.has_attachment, True)
+
+    def test_remove_attachment_while_editing(self) -> None:
+        CONST_UPLOAD_PATH_PREFIX = "/user_uploads/"
+        verona_stream_id = self.get_stream_id("Verona")
+        user_profile = self.example_user("hamlet")
+        file1 = self.create_attachment_helper(user_profile)
+
+        content = f"Init message [attachment1.txt]({file1})"
+        self.login("hamlet")
+
+        scheduled_delivery_timestamp = int(time.time() + 86400)
+
+        # Test sending with attachment
```

Comment:
The combination of names/docstrings/comments in these tests could benefit from being more clear as to what the overall plan and expected behavior is.

Here, maybe a name like `test_edit_scheduled_message_can_detach_attachment` would help, but also maybe just a bit more comments.

Reply:
Hmm, I've found it quite hard to come up with good names for these test cases. I continued with `test_edit_scheduled_message_can_detach_attachment` but I still felt like the name was off since I'd like that think off it as detach vs remove. When I think about detach, it's just removing any reference from an message, which should be doable as long as you have permission to edit. As for removing, I'd think about just getting rid of the file entirely. Which you would have to be the owner of the file to do so.

In any case I've added more comments to the test to increase clarity and provide guidance on the overall plan of the testcase.